uploadNeuroDB/NeuroDB/Database.pm

```diff
@@ -0,0 +1,325 @@
+use MooseX::Privacy;
+use DBI;
+use TryCatch;
+
+use constant DEFAULT_DB_PORT => 3306;
+
+=pod
+
+=head3 C<< new(userName => $u, databaseName => $d, hostname => $h, password => $pwd, port =>$port) >>  (constructor)
+
+Create a new instance of this class, without actually trying to connect
+to the database specified. All parameters are required except 'port', which
+defaults to 3306 if not specified. If the user name, database name or host
+name are the empty string, the constructor will call C<die>.
+
+INPUT: a set of properties for the current database:
+
+=over
+
+=item userName
+
+name of the user for the (upcoming) connection
+
+=item databaseName
+
+name of the database
+
+=item hostName
+
+name of the host on which the database resides
+
+=item password
+
+password for the (upcoming) connection
+
+=item port
+
+port used for the (upcoming) connection (defaults to 3306 if not provided)
+
+=back
+
+RETURNS: new instance of this class.
+
+=cut
+
+has 'userName'     => (is  => 'ro', isa => 'Str', required => 1);
+
+has 'databaseName' => (is  => 'ro', isa => 'Str', required => 1);
+
+has 'hostName'     => (is  => 'ro', isa => 'Str', required => 1);
+
+has 'password'     => (is  => 'ro', isa => 'Str', required => 1);
+
+has 'port'         => (is  => 'ro', isa => 'Int', default => DEFAULT_DB_PORT);
+
+has 'dbh'          => (is  => 'rw', init_arg => undef, traits => [qw/Private/]);
+
+sub BUILD {
+    my $self = shift;
+
+    die "User name cannot be the empty string.\n"     if $self->userName     eq '';
+    die "Database name cannot be the empty string.\n" if $self->databaseName eq '';
+    die "Host name cannot be the empty string.\n"     if $self->hostName     eq '';
+}
+
+=pod
+
+=head3 C<connect()>
+
+Attempts to connect to the database using the connection parameters passed
+at construction time. This method will throw a DatabaseException if the
+connection could not be established.
+
+=cut
+
+sub connect {
+    my $self = shift;
+
+    my $connectStatement = sprintf(
+         "DBI:mysql:database=%s;host=%s;port=%d;",
+         $self->databaseName,
+         $self->hostName,
+         $self->port
+    );
+
+    try {
+        $self->dbh(
+            DBI->connect(
+                $connectStatement,
+                $self->userName,
+                $self->password,
+                { PrintError => 0, RaiseError => 1, AutoCommit => 1 }
+            )
+        );
+    } catch {
+        NeuroDB::DatabaseException->throw(
+            statement    => $connectStatement,
+            args         => [],
+            errorCode    => $DBI::err,
+            errorMessage => $DBI::errstr
+        );
+    }
+}
+
+=pod
+
+=head3 C<pselect($query, @args)>
+
+Executes a select query on the database. This method will first C<prepare>
+the statement passed as parameter before sending the request to the database.
+
+INPUTS: select query to execute (containing the argument placeholders if any)
+        list of arguments to replace the placeholders with.
```

Comment:
Ok, I'll correct.

Reply:
Sorry for being picky. For INPUTS with multiple inputs, it should be documented as:

```
INPUTS:
  - select query to execute ...
  - list of arguments to replace the placeholders with ...
```

Reply:
Corrected. I noticed that when we run perldoc this formatting is lost though...

Reply:
That is weird. If you look at MRI.pm in the loris_mri_docs branch (for exemple make_jiv function), it is correctly displayed in the perdoc. Maybe something to do with the alignment?

Another note that I did not think about earlier, sorry... It might be good to create the MD file of the new scripts you created so their documentation can be stored in docs/scripts/